Javatrix.java

```diff
@@ -477,5 +477,31 @@ public Javatrix times(double scalar)
         }
         return actual;
     }
+
+    /**
+     * Gets the trace of a square matrix.
+     *
+     * @throws IllegalArgumentException if rows and
+     * columns are not the same.
+     * @return the trace of the matrix.
+     **/
+    public double trace() throws IllegalArgumentException
+    {
+        m = this.m;
+        n = this.n;
+        double ace = 0;
+        if (m != n)
+        {
+            String ex = "Rows and Columns must be same length.";
+            throw new IllegalArgumentException(ex);
+        }
+        matrix = new double[m][n];
```

Comment:
line 498 is why the function is returning 0.  You don't need to set matrix equal to a new double[][] here.